src/workerd/api/web-socket.c++

```diff
@@ -390,16 +390,9 @@ WebSocket::Accepted::Accepted(Hibernatable wsParam, Native& native, IoContext& c
 }
 
 kj::Promise<void> WebSocket::Accepted::createAbortTask(Native& native, IoContext& context) {
-  return kj::evalNow([&]() { return ws->whenAborted(); })
-      .catch_([](kj::Exception&& e) {
-    // whenAborted() is theoretically not supposed to throw, but some code paths, like
-    // AbortableWebSocket and Cap'n Proto disconnects, may end up throwing DISCONNECTED. Treat
-    // exceptions the same as if `whenAborted()` finished normally -- but log if it's not
-    // DISCONNECTED.
-    if (e.getType() != kj::Exception::Type::DISCONNECTED) {
-      LOG_EXCEPTION("webSocketWhenAborted", e);
-    }
-  }).then([this, &native]() {
+  try {
+    co_await ws->whenAborted();
+
     // Other end disconnected prematurely. We may be able to clean up our state.
     native.outgoingAborted = true;
     if (!native.isPumping && native.closedIncoming) {
```

Comment:
If I'm understanding the semantics correctly: in the previous version if this code threw a DISCONNECTED exception then that exception would get logged, with the new coroutines version this is no longer the case. I suppose this code is unlikely to throw that but I figured I'd highlight it.

Reply:
It still would. Check the catch block below.

Reply:
You mean that the new version would still log a DISCONNECTED exception in this code? I don't think so.

If a `kj::Exception` is thrown then wouldn't only the first `catch` block run? Thus if that exception is `DISCONNECTED` the exception will get silently lost. In the previous version of this code that wouldn't happen because the lambda in the `eagerlyEvaluate` block would run instead (I'm assuming that the `catch_` would only apply to the `kj::evalNow` promise).